### PR TITLE
fix: security issues

### LINKED ISSUE #321: Investigate security findings
Scans from a potential adoptor resulted in some findings. We should investigate these:

```go
pkg/sync/http_sync.go
    go.lang.security.audit.crypto.bad_imports.insecure-module-used
       Detected use of an insecure cryptographic hashing method. This method is known to be broken
       and easily compromised. Use SHA256 or SHA3 instead.
       Details: https://sg.run/l2gj
 
        69┆ hasher := sha1.New() //nolint:gosec
         ⋮┆----------------------------------------
    go.lang.security.audit.crypto.use_of_weak_crypto.use-of-sha1
       Detected SHA1 hash algorithm which is considered insecure. SHA1 is not collision resistant
       and is therefore not suitable as a cryptographic signature. Use SHA256 or SHA3 instead.
       Details: https://sg.run/XBYA
 
        69┆ hasher := sha1.New() //nolint:gosec
```

```go
pkg/sync/kubernetes/kubernetes_sync.go
     trailofbits.go.questionable-assignment.questionable-assignment
        Should `k` be modified when an error could be returned?
        Details: https://sg.run/qq6y
 
        117┆ k.client, err = client.New(clusterConfig, client.Options{Scheme: scheme.Scheme})
```

### PR DESCRIPTION
**Pull request by odubajDT**
## This PR

- fixes security findings

### Related Issues

Fixes #321

### CODE REVIEW

File: pkg/sync/kubernetes/kubernetes_sync.go [hunk 1]
```diff
@@ -65,11 +65,13 @@ func (k *Sync) Init(ctx context.Context) error {
 		return err
 	}
 
-	k.readClient, err = client.New(clusterConfig, client.Options{Scheme: scheme.Scheme})
+	readClient, err := client.New(clusterConfig, client.Options{Scheme: scheme.Scheme})
```
Comment (Kavindu-Dodan) [inline pkg/sync/kubernetes/kubernetes_sync.go:68]: Minor - the rest of the work looks good. But any reason for this change :) ?

**Review by Kavindu-Dodan — APPROVED** [review]
Apporoving and waiting for conversation to be resloved

File: pkg/sync/kubernetes/kubernetes_sync.go (hunk 1 above)
Comment (odubajDT) [reply pkg/sync/kubernetes/kubernetes_sync.go]: not really a big reason when I think about it now, reverting :)

File: pkg/sync/http/http_sync.go [hunk 2]
```diff
@@ -143,10 +142,9 @@ func (hs *Sync) fetchBodyFromURL(ctx context.Context, url string) ([]byte, error
 }
 
 func (hs *Sync) generateSha(body []byte) string {
-	hasher := sha1.New() //nolint:gosec
+	hasher := sha3.New256() //nolint:gosec
```
Comment (toddbaert) [inline pkg/sync/http/http_sync.go:145]:
```suggestion
	hasher := sha3.New256()
```

Do you know if this nolint is still needed?

**Review by toddbaert — APPROVED** [review]
TBH this is definitely a false positive security issue. The hash isn't being used here for anything security related, just to reduce un-needed changes if the HTTP contents haven't changed.

That said, if switching to sha256 save us some false positive warnings in tools I think it's a good change.

File: pkg/sync/http/http_sync.go (hunk 2 above)
Comment (odubajDT) [reply pkg/sync/http/http_sync.go:145]: not sure, but I would guess no

File: pkg/sync/kubernetes/kubernetes_sync.go (hunk 1 above)
Comment (Kavindu-Dodan) [reply pkg/sync/kubernetes/kubernetes_sync.go]: Thanks, just wanted to see if this is a common pattern or a best practice :)

**Pull request merged**